### PR TITLE
Update docs for keyword search GA

### PR DESCRIPTION
**Pull request by jtibshirani**
In addition to keyword search-related updates, this PR makes some "low hanging fruit" refactors to the search docs structure. I feel this is important to bundle with the keyword search updates so that our customers can effectively find information on the change and new syntax.

Main changes:
* Always say "keyword search" when listing search capabilities, instead of "literal search"
* In search examples, switch to `patterntype=keyword` or `regexp` instead of older types (`lucky`, `literal`, or `standard`). This was a big change, and I had to adapt some searches to quote the content "..."
* Remove prominent references to structural search, since it's disabled by default and we're trying to discourage further use. It's still discoverable through its own page under "Search Types".
* Low-hanging improvements to page structure, as I found the current one hard to follow:
  * List "Search Query Syntax" near the top
  * Move "Search Examples" under "Search Query Syntax"
  * Rename "Working with Search" to "Advanced Features"

Closes SPLF-107

### CODE REVIEW

**Comment by jtibshirani** [comment]
I think the page structure could still be improved a lot. I think "Search Types" should not really be its own page, as it's not a true concept in search. Maybe in a follow-up, we should delete it and move all its subpages, either to "Search Query Syntax" or "Advanced Features" as appropriate.

File: docs/code-search/queries/language.mdx [hunk 1]
```diff
@@ -43,12 +43,6 @@ For example,
 - [`foo.*bar.*baz`](https://sourcegraph.com/search?q=foo.*bar.*baz&patternType=regexp)
 - [`"foo bar"`](https://sourcegraph.com/search?q=%22foo+bar%22&patternType=regexp)
 
-Perform a structural search. See our [dedicated documentation](/code-search/types/structural) to learn more about structural search.
```
Comment (jtibshirani) [inline docs/code-search/queries/language.mdx:46]: Removed this prominent structural search example

File: docs/code-search/queries/index.mdx [hunk 2]
```diff
@@ -28,29 +25,13 @@ Keyword search matches individual terms anywhere in the document or the filename
 | [`foo OR bar`](https://sourcegraph.com/search?q=context:global+foo+OR+bar&patternType=keyword)                                            | Match documents containing `foo` _or_ `bar` anywhere in the document.                                                            |
 
 <Callout type="tip">Matching is case-insensitive by default (toggle the `Aa` button to change).</Callout>
+<Callout type="note">In version 5.4, `keyword` replaced `standard` as the new default pattern type.</Callout>
 
-<Callout type="tip">Keyword search can be enabled for a Batch Change by adding [patternType:keyword](/code-search/queries/language#pattern-type) to the query in [a batch spec](/batch-changes/quickstart).</Callout>
-
-<Callout type="tip">[patternType:keyword](/code-search/queries/language#pattern-type) is supported in [Code Monitors](/code_monitoring/quickstart) and [Code Insights](/code_insights/quickstart). When creating a monitor or an insight from a search, it will be automatically populated with `patternType:keyword`. When building one by hand, be sure to include `patternType:keyword` if you want keyword search.</Callout>
-
-### Standard search
```
Comment (jtibshirani) [inline docs/code-search/queries/index.mdx:36]: Removed the "Standard search" and "Structural search" sections to keep these high-level docs simple

Comment (stefanhengl) [reply docs/code-search/queries/index.mdx:36]: It feels very good to remove these sections here, but maybe we should keep structural search and move it somewhere else?

**Review by stefanhengl — APPROVED** [review]

**Comment by stefanhengl** [comment]
@MaedahBatool This PR relevant for the upcoming July 10th release. Should the PR merge into a special release branch instead of into main?

File: docs/code-search/queries/index.mdx [hunk 3]
```diff
@@ -61,45 +42,37 @@ Clicking the (.*) toggle interprets all search patterns as regular expressions.
 
 <Callout type="note">As in Standard search, we support [RE2 syntax](https://golang.org/s/re2syntax). Matching is case-insensitive by default (toggle the `Aa` button to change).</Callout>
```
Comment (stefanhengl) [inline docs/code-search/queries/index.mdx:43]:
```suggestion
<Callout type="note">We support [RE2 syntax](https://golang.org/s/re2syntax). Matching is case-insensitive by default (toggle the `Aa` button to change).</Callout>
```

File: docs/code-search/queries/index.mdx (hunk 2 above)
Comment (stefanhengl) [reply docs/code-search/queries/index.mdx:36]: never mind, I saw we still have a dedicated page for structural search.

**Comment by MaedahBatool** [comment]
> @MaedahBatool This PR is relevant for the upcoming July 10th release. Should the PR merge into a special release branch instead of into main?

@stefanhengl thank you for opening this PR. I have pointed it to the docs working branch. If this is final from your end let's merge it into the working branch.
